zerver/forms.py

```diff
@@ -262,6 +262,19 @@ def email_is_not_disposable(email: str) -> None:
 class RealmCreationForm(forms.Form):
     # This form determines whether users can create a new realm.
     email = forms.EmailField(validators=[email_not_system_bot, email_is_not_disposable])
+    realm_subdomain = forms.CharField(max_length=Realm.MAX_REALM_SUBDOMAIN_LENGTH, required=False)
+    realm_type = forms.TypedChoiceField(
+        coerce=int, choices=[(t["id"], t["name"]) for t in Realm.ORG_TYPES.values()]
+    )
+    realm_name = forms.CharField(max_length=Realm.MAX_REALM_NAME_LENGTH)
+
+    def clean_realm_subdomain(self) -> str:
+        subdomain = self.cleaned_data["realm_subdomain"]
+        if "realm_in_root_domain" in self.data:
+            subdomain = Realm.SUBDOMAIN_FOR_ROOT_DOMAIN
+
+        check_subdomain_available(subdomain)
+        return subdomain
```

Comment:
Does it make sense to change the registration form to subclass this or something to avoid duplication of the checks for fields that appear in both?

Reply:
I have tried using a new class `RealmDetailsForm` which is used as subclass by both `RealmCreationForm` and `RegistrationForm`.

I have not directly used `RealmCreationForm` as subclass for `RegistrationForm` since `email` is not a field in `RegistrationForm` and we would need to set `email = None` in `RegistrationForm` in that case.